src/serac/numerics/solver_config.hpp

```diff
@@ -103,9 +103,12 @@ enum class LinearSolver
  */
 enum class NonlinearSolver
 {
-  MFEMNewton,               /**< Newton-Raphson */
-  KINFullStep,              /**< KINFullStep */
-  KINBacktrackingLineSearch /**< KINBacktrackingLineSearch */
+  Newton,                    /**< MFEM-native Newton-Raphson */
+  LBFGS,                     /**< MFEM-native Limited memory BFGS */
+  KINFullStep,               /**< KINSOL Full Newton (Sundials must be enabled) */
+  KINBacktrackingLineSearch, /**< KINSOL Newton with Backtracking Line Search (Sundials must be enabled) */
+  KINPicard,                 /**< KINSOL Picard (Sundials must be enabled) */
+  KINFP                      /**< KINSOL Fixed Point (Sundials must be enabled) */
```

Comment:
I think these methods only work for specific kinds of nonlinear equations, so they are likely not suitable for solving residual equations in general (unless KINSOL is doing something I don't understand)

Reply:
Also, I think Jorge-Luis mentioned earlier that Newton and KINFullStep were producing significantly different results, which seems unexpected

Reply:
That definitely seems weird. I'll investigate this.